portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -80,21 +85,26 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
             e.printStackTrace();
         }
 
-        long timestamp = System.currentTimeMillis();
         String username = getAttribute(json, "username", "Anonymous");
         String comment = getAttribute(json, "comment", "...");
+        long timestamp = System.currentTimeMillis();
+        double sentiment = analyzeSentiment(comment);
+        Emotion emotion =
+            sentiment >= -1 && sentiment <= -0.3 ? Emotion.Mad :
+            sentiment > -0.3 && sentiment <= 0.3 ? Emotion.Neutral :
+            Emotion.Happy;
```

Comment:
this is an FYI comment, so please don't worry about changing the code. I'm just including it for educational purposes.

The idiomatic Java way to implement this would be to add a `static` method to the `Emotion` enum, something like:
```java
private enum Emotion {
    Happy,
    Neutral,
    Mad;

    static Emotion fromSentiment(double sentiment) {
        return sentiment >= -1 && sentiment <= -0.3 ? Mad :
               sentiment > -0.3 && sentiment <= 0.3 ? Neutral :
               Happy;
    }
}
```

and then this line would just become

```java
Emotion emotion = Emotion.fromSentiment(sentiment);
```

Reply:
Oh that is totally better!
I didn't know you could add methods to enums. Will definitely look into that!